### ISSUE TITLE
BUG - 3 - All - Demographics data not loaded first time opening Personal Information screen

### ISSUE THREAD

**Issue by theodur** [department-of-veterans-affairs/va-mobile-app#6980]
## What happened?

When the Personal Information screen is opened for the first time, demographics information isn't loaded. The gender identity and preferred name fields are blank. They only load after navigating into one of the update screens, and navigating back out. It seems like this caused by the `staleTime` cache setting that was set in the [contact info endpoint PR](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6733). If `staleTime` is not set, demographics information loads as expected. This doesn't seem to be impacting other queries though. 

## Specs:

- Device:
- OS:
- User Account:
- Accessibility State:

## Steps to Reproduce

Navigate to the Personal Information screen after recently signing in

## Desired behavior

Demographics information should be loaded when the Personal Information screen is loaded.

## Acceptance Criteria

- [ ] Demographics data is loaded when the Personal Information screen is loaded

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: High Low 
- Frequency: High Low 

- 3 - Low
- 2 - High
- 1 - Critical

## Screen shot(s) and additional information

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [ ] Labels added (front-end, back-end, relevant feature, accessibility, etc)
- [ ] Estimate of 1 added (for front-end tickets)
- [ ] Added either to the relevant feature epic (if found during new feature implementation), or the [bug epic](https://app.zenhub.com/workspaces/va-mobile-product-view-610035bc5395bb000e62e529/issues/gh/department-of-veterans-affairs/va-mobile-app/5367)

**Comment by TKDickson** [comment]
Also impacts navigating to the Letters screen right after login (because address shows there), I think? Something to keep an eye on at least @theodur @rbontrager

**Comment by theodur** [comment]
@TKDickson Since this is related to the gender identity options endpoint, and not the new contact info endpoint, this only impacted screens that referenced gender identity, primarily the Personal Information screen. I just updated the ticket title to reflect that, because initially I thought demographics data wasn't being fetched, which wasn't the case.

For reference, the root of the issue was I accidentally set the initial data for the gender identity options to be an empty object in [this PR](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6627/files#diff-918f2508e333c5782af7f24120684dcd9c3662516d967242801107e7042fbf1dR35). This worked fine when the query cache was configured to immediately mark queries as stale (React Query's default) because it meant the app would always fetch the latest data from the API whenever gender identity options were referenced by a screen. The problem arose from the [contact info endpoint PR](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6733) because I updated the query cache to wait 5 seconds before marking queries as stale to solve an [issue](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6733#discussion_r1339362211) with certain screens refetching from the API too frequently. The 5-second buffer meant the initial data for the gender identity options (which was an empty object) was marked as fresh for 5 seconds, so it wasn't being fetched from the API the first time the Personal Information screen was loaded, which is why the gender identity wasn't showing up, because without the options, the screen can't map the user's gender identity to the correct option to display. The initial data didn't need to be set for the gender identity options query, that was just a change I made during testing that I forgot to clean up 😅

TLDR; The gender identity options weren't being fetched from the API on the initial screen load due to: 
- The data being set as an empty object from a previous PR 
- The cache stale time being increased to 5 seconds in a more recent PR

**Comment by TKDickson** [comment]
@theodur that makes a lot of sense, thanks for the explanation!
